src/processing/resources/page.js

```diff
@@ -33,6 +34,22 @@ class PageProcessor extends ResourceProcessorBase {
         this.serializer = new parse5.Serializer();
     }
 
+    _createRestoreStoragesScript (storageKey, storages) {
+        const scriptStr = dedent(`
+            <script type="text/javascript" class="${ SHADOW_UI_CLASSNAME.selfRemovingScript }">
```

Comment:
There are two self removing scripts ( `BODY_CREATED_EVENT_SCRIPT` and restore storages script) with shared client code: tag, self-executing function and remove scripts parts.
Let's create the utility function `createSelfRemovingScript (content)` and use it for both scripts.